src/magic.cpp

```diff
@@ -225,6 +243,24 @@ void spell_type::load( JsonObject &jo, const std::string & )
     optional( jo, was_loaded, "casting_time_increment", casting_time_increment, 0.0f );
 }
 
+static bool spell_infinite_loop_check( std::set<spell_id> &spell_effects, const spell_id &sp )
```

Comment:
This check is failing, though it's passing on travis since it isn't loading the mod.
It's adding lightning blast to the spell_effects set, then it *visits* lightning_blast, which fails since it's on the list already.

What you want to do is at each node, only consider the previously visited nodes, this requires you to make copies of your set, but unless someone makes a spell with hundreds of chained effects, that doesn't really matter.

In short, pass spell_effects by copy instead of by reference, and only ever insert the *current* spell effect into the set before recursing to the invoked spell effects.